cmd/iam.go

```diff
@@ -1617,59 +1623,74 @@ func (sys *IAMSys) IsAllowedServiceAccount(args iampolicy.Args, parent string) b
 func (sys *IAMSys) IsAllowedSTS(args iampolicy.Args) bool {
 	// If it is an LDAP request, check that user and group
 	// policies allow the request.
-	if userIface, ok := args.Claims[ldapUser]; ok {
-		var user string
-		if u, ok := userIface.(string); ok {
-			user = u
-		} else {
-			return false
-		}
+	if sys.usersSysType == LDAPUsersSysType {
+		if userIface, ok := args.Claims[ldapUser]; ok {
```

Comment:
I think it is better to have a function isAllowedSTSForLdap() where it contains the code of this block

```
if sys.usersSysType == LDAPUsersSysType {
    return sys.isAllowedSTSForLdap()
}
```

Reply:
Yes sure but that is not the purpose of this PR @vadmeste